assets/chezmoi.io/docs/reference/templates/functions/setValueAtPath.md

```diff
@@ -0,0 +1,14 @@
+# `setValueAtPath` *path* *value* *dict*
+
+`setValueAtPath` modifies *dict* to set the value at *path* to *value* and
+returns *dict*. *path* can be either a string containing a `.`-separated list of
+keys or a list of keys. It is an alternative to [sprig's `set`
+function](http://masterminds.github.io/sprig/dicts.html) with a different
+argument order that supports pipelining.
```

Comment:
We should probably document that `setValueAtPath "root.key.bar.baz" "quux"` will *absolutely* error out given the following input:

```json
{
  "root": {
    "key": {}
  }
}
```

It *could* be nice to have a `createOrSetValueAtPath` variant which would create `"bar": { "bar": "quux" }`.

Reply:
This doesn't error, it creates the structure as needed:

```console
$ ./chezmoi --version
chezmoi version dev, commit 27c4ed8df22885d9ea49b4c9d0829d9b2097aebe, built at 2022-11-01T20:18:58Z
$ ./chezmoi execute-template '{{ dict "root" (dict "key" dict) | toPrettyJson }}'
{
  "root": {
    "key": {}
  }
}
$ ./chezmoi execute-template '{{ dict "root" (dict "key" dict) | setValueAtPath "root.key.bar.baz" "quux" | toPrettyJson }}'
{
  "root": {
    "key": {
      "bar": {
        "baz": "quux"
      }
    }
  }
}                                                                                                                                                                                                                                                                                                                                         
```

Reply:
I've updated the documentation to document the create behavior. I don't think we need to rename the function as the name is already quite unwieldy.

Reply:
We only needed the function renamed if the behaviour was halt-y.